net/rhp/session.go

```diff
@@ -133,48 +134,53 @@ func (s *Session) WriteRequest(rpcID Specifier, req ProtocolObject) error {
 // ReadID reads an RPC request ID. If the renter sends the session termination
 // signal, ReadID returns ErrRenterClosed.
 func (s *Session) ReadID() (rpcID Specifier, err error) {
-	defer wrapErr(&err, "ReadID")
 	err = s.readMessage(&rpcID, 16)
 	if rpcID == loopExit {
 		err = ErrRenterClosed
+	} else if err != nil {
+		err = fmt.Errorf("ReadID: %w", err)
 	}
 	return
 }
 
 // ReadRequest reads an RPC request using the new loop protocol.
 func (s *Session) ReadRequest(req ProtocolObject, maxLen uint64) (err error) {
-	defer wrapErr(&err, "ReadRequest")
-	return s.readMessage(req, maxLen)
+	err = s.readMessage(req, maxLen)
+	if err != nil {
+		err = fmt.Errorf("ReadRequest: %w", err)
+	}
+	return
```

Comment:
could define an `addContext` helper for these callsites as well.

We also don't need to use named returns for these functions anymore.